x/ibc/07-tendermint/types/misbehaviour_handle.go

```diff
@@ -0,0 +1,141 @@
+package types
+
+import (
+	"time"
+
+	tmtypes "github.com/tendermint/tendermint/types"
+
+	"github.com/cosmos/cosmos-sdk/codec"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+	clientexported "github.com/cosmos/cosmos-sdk/x/ibc/02-client/exported"
+	clienttypes "github.com/cosmos/cosmos-sdk/x/ibc/02-client/types"
+)
+
+// CheckMisbehaviourAndUpdateState determines whether or not two conflicting
+// headers at the same height would have convinced the light client.
+//
+// NOTE: consensusState1 is the trusted consensus state that corresponds to the TrustedHeight
+// of misbehaviour.Header1
+// Similarly, consensusState2 is the trusted consensus state that corresponds
+// to misbehaviour.Header2
+func (cs ClientState) CheckMisbehaviourAndUpdateState(
```

Comment:
Think this is debatable hygiene-wise since `client_state.go` would become humongous. Perhaps leaving as-is and using docs to point people in the right direction is the better approach. Definitely something to discuss and come to agreement on, either way out-of-scop for this PR imo.

cc: @fedekunze

Reply:
I agree with @AdityaSripal. The file would be too large if we add it